src/http/http_client.f90

```diff
@@ -263,6 +229,7 @@ function client_header_callback(ptr, size, nmemb, client_data) bind(c)
             h_value = h_value( : len(h_value)-2)
             if(len(h_value) > 0 .and. len(h_key) > 0) then
                 call append_header(response%header, h_key, h_value)
+                ! response%header = [response%header, header_type(h_key, h_value)]
```

Comment:
```
response%header = [response%header, header_type(h_key, h_value)]
```
While Using this syntax i am getting below error : 
```
In file '././src/http/http_client.f90', around line 234: Error reallocating to 11692440854926016608 bytes: Cannot allocate memory

Error termination. Backtrace:
#0  0x7f8450d13ad0 in ???
#1  0x7f8450d14649 in ???
#2  0x7f8450d149bc in ???
#3  0x7f84510d7579 in client_header_callback
        at ././src/http/http_client.f90:233
#4  0x7f8451021701 in ???
#5  0x7f8451032b84 in ???
#6  0x7f84510161f3 in ???
#7  0x7f845101941d in ???
#8  0x7f8450ff5e52 in ???
#9  0x7f84510d9f33 in __http_client_MOD_client_get_response
        at ././src/http/http_client.f90:113
#10  0x7f84510dc993 in __http_client_MOD_new_request
        at ././src/http/http_client.f90:56
#11  0x7f84510d2376 in response_header
        at example/response_header.f90:19
#12  0x7f84510d3815 in main
        at example/response_header.f90:2
<ERROR> Execution for object " response_header " returned exit code  1
<ERROR> *cmd_run*:stopping due to failed executions
STOP 1
``` 
the error is not consistent sometime it occurs, and sometimes not

Reply:
My guess is the array allocation in `append_header()`, called from `client_header_callback()`.

In `append_header()`, you can use the intrinsic `move_alloc()` to reallocate the header array. Also, `http_client.f90` is probably not the right place for the `append_header()` routine as it updates the header from `http_header.f90`. Prefixes added to all procedures which still lack them would make that clearer.

You can debug the app in GDB:

```
$ sudo gdb ./app
(gdb) run
```

After the seg fault, show the backtrace:
```
(gdb) bt
```

You may want to compile against a libcurl library that has debug flags set. The backtrace would then show all called cURL procedures as well.

Reply:
Thanks, @interkosmos, i will migrate `append_header()` procedure to `http_header.f90` and also I will try to debug it, but actually `append_header()` procedure is working fine, but when i use the below code to append the header, i am getting error. 👇 
```
response%header = [response%header, header_type(h_key, h_value)]
```

Reply:
In my opinion, it would be beneficial to migrate the `append_header()` procedure to `http_response.f90` instead of `http_header.f90`. This is because `http_header.f90` only contains the definition of `header_type`, which is limited to storing two character strings for the `key` and `value`. On the other hand, `http_request.f90` uses the `header_type` array to store actual request headers. Therefore, migrating the `append_header()` procedure to `http_response.f90` would be more appropriate. What do you think?

Reply:
Because the array is already allocated. You are trying to copy one header type more into the array than there is room for.

Reply:
Alright, I have used the same syntax to set the default request header as follows:
```fortran
! Set default request headers.
request%header = [header_type('user-agent', 'fortran-http/1.0.0')]
if(present(header)) then 
    request%header = [header, request%header]
end if
```
Is this also a concern?

Reply:
Yes. You have to reallocate (“grow”) the array first with `move_alloc()`.

Reply:
Are you sure about this? Reallocation on assignment normally just works. I haven't tried it but I suspect the issue is elsewhere. `move_alloc` should work fine as well.

Reply:
Uhh, I didn’t expect that to work:
```fortran
program main
    implicit none
    integer, allocatable :: foo(:)
    foo = [ 1, 2 ]
    foo = [ foo, 3 ]
end program main
```